common/djangoapps/track/backends/adaptive_learning.py

```diff
@@ -0,0 +1,83 @@
+""" Event tracker backend that sends relevant data to adaptive learning service. """
+
+from __future__ import absolute_import
+
+import logging
+
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+from track.backends import BaseBackend
+from xmodule.library_content_module import AdaptiveLibraryContentModule
+from xmodule.modulestore.django import modulestore
+
+
+log = logging.getLogger(__name__)
+
+
+def _is_problem_check(event):
+    """
+    Return True if `event` is of type `problem_check`, else False.
+    """
+    event_type = event.get('event_type')
+    return event_type == 'problem_check'
+
+
+def _get_course(event):
+    """
+    Retrieve course corresponding to course ID mentioned in `event` from DB
+    and return it.
+    """
+    context = event.get('context')
+    course_id = context.get('course_id')
+    course_key = CourseKey.from_string(course_id)
+    course = modulestore().get_course(course_key)
+    return course
+
+
+def _get_block_id(event):
+    """
+    Return ID of block that `event` belongs to.
+    """
+    context = event.get('context')
+    module = context.get('module')
+    usage_key_string = module.get('usage_key')
+    usage_key = UsageKey.from_string(usage_key_string)
+    block_id = usage_key.block_id
+    return block_id
+
+
+def _get_user_id(event):
+    """
+    Return ID of user that triggered `event`.
+    """
+    context = event.get('context')
+    user_id = context.get('user_id')
+    return user_id
+
+
+def _get_success(event):
+    """
+    Return success information from `event`.
+    """
+    event = event.get('event')
+    success = event.get('success')
```

Comment:
@itsjeyd and also there's a lot of this pattern:

    part1 = event.get('part1')
    part2 = event.get('part2')

Maybe it would be advantageous to have something like

    def get_from_event(event, path, default):
        parts = path.split('.')
        for part in parts[:-1]:
            current = event.get(part, {})
        return current.get(parts[-1], default)

And use it like

    get_from_event(event, 'event.success', None)